### PR TITLE
Adding ceph config command option to cluster Close #332

### LINKED ISSUE #332: [doc] 2.1.1.1  Adding a Private Network to a Running Cluster
### Type of report
   - [ ] bug

### Source Document
2.1.1.1  Adding a Private Network to a Running Cluster:

https://susedoc.github.io/doc-ses/master/single-html/ses-deployment/#storage-bp-net-private

### Summary

There is another way in SES7 how to edit Ceph configuration. Instead of editing ceph.conf directly there is a command `ceph config` that can do the task. In thi particular example:
`ceph config set global cluster_network 10.0.0.0/24`
And there is no need to restart the services.

### PR DESCRIPTION
**Pull request by asettle**

### CODE REVIEW

**Comment by jan--f** [comment]
Hmm I don't think this works via `ceph config`. Not sure but I suspect this would be handled via `ceph-salt`, ping @ricardoasmarques .

**Comment by asettle** [comment]
> Hmm I don't think this works via `ceph config`. Not sure but I suspect this would be handled via `ceph-salt`, ping @ricardoasmarques .

I've also pinged Ricardo in the original issue - sorry for the confusion

**Comment by ricardoasmarques** [comment]
I'm not familiar with the `cluster_network` setting in particular, but in `ceph-salt` you can only specify `ceph.conf` settings that will be used during the bootstrap (settings that will be used in `cephadm bootstrap --config <ceph_conf>` command)

For instance:

```
# ceph-salt config /cephadm_bootstrap/ceph_conf add global
Section added.

# ceph-salt config /cephadm_bootstrap/ceph_conf/global set cluster_network 10.10.100.0/24
Parameter set.

# ceph-salt config /cephadm_bootstrap/ceph_conf ls
o- ceph_conf ................................................ [...]
  o- global ................................................. [...]
    o- cluster_network ........................... [10.10.100.0/24]
```

But again, I don't know if this will have any effect when used during cluster bootstrap.

Maybe someone more familiar with `ceph.conf`/`cluster_network`/`cephadm bootstrap` can help here?

**Comment by asettle** [comment]
> Maybe someone more familiar with `ceph.conf`/`cluster_network`/`cephadm bootstrap` can help here?

@sebastian-philipp ?

File: xml/deployment_hwrecommend.xml [hunk 1]
```diff
@@ -161,6 +161,11 @@
 <screen>&prompt.root;systemctl start ceph.target</screen>
      </step>
     </procedure>
+    <para>
+      Alternatively, you can also use the <command>ceph config</command> to perform
+      the same task:
+    </para>
+<screen>&prompt.cephuser;ceph config set global cluster_network <replaceable>IP_ADDRESS</replaceable></screen>
```
Comment (sebastian-philipp) [inline xml/deployment_hwrecommend.xml:168]:
```suggestion
<screen>&prompt.cephuser;ceph config set mon cluster_network <replaceable>IP_ADDRESS</replaceable></screen>
```

source: https://docs.ceph.com/docs/master/cephadm/install/#deploy-additional-monitors-optional

File: xml/deployment_hwrecommend.xml [hunk 2]
```diff
@@ -161,6 +161,11 @@
 <screen>&prompt.root;systemctl start ceph.target</screen>
      </step>
     </procedure>
+    <para>
+      Alternatively, you can also use the <command>ceph config</command> to perform
+      the same task:
+    </para>
+<screen>&prompt.cephuser;ceph config set mon cluster_network <replaceable>IP_ADDRESS</replaceable></screen>
```
Comment (asettle) [inline xml/deployment_hwrecommend.xml:168]: <screen>&prompt.cephuser;ceph config set global cluster_network <replaceable>IP_ADDRESS</replaceable></screen>

**Comment by asettle** [comment]
Closing this and the bug report. Unable to verify whether or not this actually works (I tested also, and was unable to achieve the desired result).

**Pull request closed without merging**